### PR TITLE
Add very-simple-smtps package

### PR DESCRIPTION
**Pull request by matthewT53**
Specify library name and version:  **very-simple-smtps/1.0.0**

I would like to add a package for a library that allows a C++ application to connect to a SMTPS server to send an email with binary attachments. Its been very useful to me in a few projects and I would like to share it with the community! :smile: 

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/very-simple-smtps/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+from conan.tools.meson import Meson, MesonToolchain
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+#
+# INFO: Please, remove all comments before pushing your PR!
```
Comment (uilianries) [inline recipes/very-simple-smtps/all/conanfile.py:18]: Please, follow the comment instruction. Any comment in the recipe should really useful, like a related issue number, or why some configuration is not supported.

File: recipes/very-simple-smtps/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("very-simple-smtps is only supported by Linux")
+
+        if self.settings.compiler == "clang" and self.settings.compiler.libcxx == "libc++":
+            raise ConanInvalidConfiguration("very-simple-smtps can not use libc++")
```
Comment (uilianries) [inline recipes/very-simple-smtps/all/conanfile.py:79]: Validations should be moved to `validate()` method.

File: recipes/very-simple-smtps/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": False,
+    }
+
+    required_packages = [
+        "libcurl/8.0.1",
+        "doctest/2.4.11",
+    ]
```
Comment (uilianries) [inline recipes/very-simple-smtps/all/conanfile.py:47]: Please, move it directly to `requirementes()` method, let's keep it simple.

File: recipes/very-simple-smtps/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+            "gcc": "9",
+            "clang": "9",
+            "apple-clang": "10",
+        }
+
+    # no exports_sources attribute, but export_sources(self) method instead
+    # this allows finer grain exportation of patches per version
+    def export_sources(self):
+        export_conandata_patches(self)
+
```
Comment (uilianries) [inline recipes/very-simple-smtps/all/conanfile.py:66]: Please, remove it. There are no patches to be applied, so this part of the template has no use here.

File: recipes/very-simple-smtps/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        tc.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        # remove bundled xxhash
+        # rm(self, "whateer.*", os.path.join(self.source_folder, "lib"))
+        # replace_in_file(self, os.path.join(self.source_folder, "meson.build"), "...", "")
+
+    def build(self):
+        self._patch_sources()  # It can be apply_conandata_patches(self) only in case no more patches are needed
```
Comment (uilianries) [inline recipes/very-simple-smtps/all/conanfile.py:142]:
```suggestion
    def build(self):
```
No patches to be applied. Let's keep it simpler, please.

**Review by uilianries — COMMENTED** [review]
@matthewT53 Thank you for your PR! Please, let's simplify the current recipe, I see many things from the CCI template that are not used here.

File: recipes/very-simple-smtps/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+#
+# INFO: Please, remove all comments before pushing your PR!
+#
```
Comment (danimtb) [inline recipes/very-simple-smtps/all/conanfile.py:19]:
```suggestion
```

File: recipes/very-simple-smtps/all/test_package/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.layout import basic_layout
+from conan.tools.meson import Meson
+import os
+
+
+# It will become the standard on Conan 2.x
```
Comment (danimtb) [inline recipes/very-simple-smtps/all/test_package/conanfile.py:8]:
```suggestion
```

**Review by danimtb — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**